src/raw/array.rs

```diff
@@ -275,7 +275,10 @@ impl<'a> Iterator for RawArrayIter<'a> {
 
     fn next(&mut self) -> Option<Result<RawBsonRef<'a>>> {
         match self.inner.next() {
-            Some(Ok((_, v))) => Some(Ok(v)),
+            Some(Ok(elem)) => match elem.value() {
```

Comment:
Can this just be `Some(Ok(elem)) => Some(elem.value())` ?

Reply:
yup!